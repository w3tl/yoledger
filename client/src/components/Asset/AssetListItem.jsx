import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { assetPropType } from './propTypes';

function AssetListItem({ url, asset }) {
  const { id, name } = asset;
  return (
    <li key={name}>
      {url ? (
        <Link to={{ pathname: `${url}/view`, state: { id } }}>{name}</Link>
      ) : (
        name
      )}
    </li>
  );
}

AssetListItem.propTypes = {
  url: PropTypes.string,
  asset: assetPropType.isRequired,
};

AssetListItem.defaultProps = {
  url: null,
};

export default AssetListItem;
