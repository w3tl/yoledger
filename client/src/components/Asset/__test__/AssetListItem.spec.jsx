import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AssetListItem from '../AssetListItem';

jest.mock('react-router-dom', () => ({ // eslint-disable-next-line react/prop-types
  Link: ({ to, children }) => <a href={to}>{children}</a>,
}));

configure({ adapter: new Adapter() });

const asset = {
  id: '1',
  name: 'Bank Card',
};

describe('AssetListItem', () => {
  it('should render correctly', () => {
    const output = shallow(
      <AssetListItem asset={asset} />,
    );
    expect(output.debug()).toMatchSnapshot();
  });

  it('should render Link when run with router', () => {
    const output = mount( // COMBAK: Check link and do not render component
      <AssetListItem asset={asset} url="/home" />,
    );
    expect(output.debug()).toMatchSnapshot();
  });
});
