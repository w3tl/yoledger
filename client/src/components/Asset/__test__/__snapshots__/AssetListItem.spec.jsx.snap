// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetListItem should render Link when run with router 1`] = `
"<AssetListItem asset={{...}} url=\\"/home\\">
  <li>
    <Link to={{...}}>
      <a href={{...}}>
        Bank Card
      </a>
    </Link>
  </li>
</AssetListItem>"
`;

exports[`AssetListItem should render correctly 1`] = `
"<li>
  Bank Card
</li>"
`;
